Source/WebKit/Shared/Cocoa/CoreIPCNSShadow.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if PLATFORM(COCOA)
+
+#include "ArgumentCodersCocoa.h"
+#include "CoreIPCSecureCoding.h"
+#include <WebCore/Color.h>
+#include <wtf/RetainPtr.h>
+
+OBJC_CLASS NSShadow;
+
+namespace WebKit {
+
+class CoreIPCSecureCoding;
+
+class CoreIPCNSShadow {
+public:
+    CoreIPCNSShadow(NSShadow *);
+    CoreIPCNSShadow(const RetainPtr<NSShadow>& shadow)
+        : CoreIPCNSShadow(shadow.get())
+    {
+    }
+
+    CoreIPCNSShadow(CGSize shadowOffset, CGFloat shadowBlurRadius, std::optional<WebCore::Color>&& shadowColor)
+        : m_shadowOffset(shadowOffset)
+        , m_shadowBlurRadius(shadowBlurRadius)
+        , m_shadowColor(WTFMove(shadowColor))
+    {
+    }
+
+    RetainPtr<id> toID() const;
+
+private:
+    friend struct IPC::ArgumentCoder<CoreIPCNSShadow, void>;
+
+    CGSize m_shadowOffset;
+    CGFloat m_shadowBlurRadius;
+    std::optional<WebCore::Color> m_shadowColor;
```

Comment:
Should I just use `NSColor *` here since that's already a supported type so I'm not re-implementing conversion between `NSColor *` and `WebCore::Color`?

Feels like I'm missing something else, too.